packages/vite/src/node/plugins/asset.ts

```diff
@@ -210,6 +210,10 @@ export function checkPublicFile(
     return
   }
   const publicFile = path.join(publicDir, cleanUrl(url))
+  if (!publicFile.startsWith(publicDir)) {
+    // can happen if e.g. url starts with '/../node_modules/', see #11145
```

Comment:
nitpick: personally I would simplify this comment a bit. it's not such a complicated thing that we need a link to the issue and it doesn't really matter what comes after `../`. also should probably capitalize URL
```suggestion
    // can happen if URL starts with '../'
```